### PR TITLE
more powerful notebook image

### PR DESCRIPTION
**Pull request by rabernat**
I would like to have more memory in the notebook pods, primarily to support very large dask graphs getting sent to the scheduler. I originally proposed these changes at pangeo-data/helm-chart#7, but it was suggested they belong here instead.

Two points raised by @mrocklin in that PR discussions:

> - I think that our VM's cpu:memory ratio is something like 1:3G . I recommend sticking to this. Leaving CPU on the table at the pod-level doesn't reduce our bill at all.
> - Currently none of our node pools have VMs large enough to run 16GB pods. I suggest that we increase the size of the nodes in the pre-emptible pool. You should be able to do this in the cloud console.

I really don't know what I'm doing here, so feedback is needed.

### CODE REVIEW

File: gce/jupyter-config.yaml [hunk 1]
```diff
@@ -14,6 +14,12 @@ jupyterhub:
           mountPath: /dev/fuse
     cloudMetadata:
       enabled: true
+     cpu:
```
Comment (jacobtomlinson) [inline gce/jupyter-config.yaml]: Looks like your indentation here is off by one space.

**Comment by rabernat** [comment]
If I merge this, what steps do I have to take to re-deploy the cluster with the new config? Is this still accurately described by the [update instructions](https://github.com/pangeo-data/pangeo/blob/master/gce/update.md)?

I an in serious need of more RAM on my scheduler.

**Comment by jacobtomlinson** [comment]
Those are the instructions I followed last time. But then again I mangled the cluster. @mrocklin do you have any quick thoughts?

**Comment by mrocklin** [comment]
My hope is that we could change the preemptible node pool without destroying the cluster.

If we do that's ok too.  The biggest cost for me is that at I'll need to ask Anaconda IT (or someone else with privileges) to route the pangeo.pydata.org domain to the new IP address.

**Comment by mrocklin** [comment]
This is fine though as long as we don't do it every day.

**Comment by rabernat** [comment]
In that case, let's make sure we do it right. Are there any other updates or changes we need to make to the cluster in the meantime?

**Comment by rabernat** [comment]
> My hope is that we could change the preemptible node pool without destroying the cluster.

Sorry, I may have misunderstood. How do we change the node pool without destroying the cluster?

**Comment by mrocklin** [comment]
If you click the little trash icon next to the preemptible pool while in edit mode you can delete the pool.  You can then create another.  

image


I can also do this if you like, but thought I'd make a record here in the "teach a man to fish" approach.

**Comment by jhamman** [comment]
Don't we also need to update the worker-template to reflect the larger vms?

https://github.com/pangeo-data/pangeo/blob/4d6d41d62dfbbe2ac06dcaa7f74ccb4e03d66e6e/gce/notebook/worker-template.yaml#L8-L29

@mrocklin - do we need to set the memory thread limits in the worker template? It would be nice if the workers could, by default, use the resources available in the VM. (BTW, this would fix the AWS pangeo too, since the vms are slightly smaller than the gce vms)

**Comment by mrocklin** [comment]
We can change the memory in config files, yes.  However we also need VMs
that have at least that much memory.  Currently I think that our VMs are
small enough such that no single VM is able to hold a 16GB memory pod, so
we will need to increase that as well.

We can update the worker-template or not.  If pods are smaller than VMs
then Kubernetes will pack many pods into a single VM.

**Comment by rabernat** [comment]
The specific issue I have is that the dask graphs for large datasets cause the scheduler (which runs in the notebook pod) to run out of memory.  4 GB is not enough for the calculations I want to do. I don’t necessarily need to increase the memory on the workers—I can just get more workers if needed. But there is only one scheduler.

**Comment by jhamman** [comment]
Thanks for the clarification @rabernat, I was thinking you needed larger workers. 

> We can update the worker-template or not.  If pods are smaller than VMs then Kubernetes will pack many pods into a single VM.

Okay, I think I'm following. The challenge we had on the AWS deployment was the VMs were smaller than what was listed in the worker template. So to get things to work there, you needed to edit the worker template manually. Or recreate the entire worker docker image.

**Comment by rabernat** [comment]
The type of machine in our preemptible pool is `n1-standard-4 (4 vCPUs, 15 GB memory)`

We are currently loosely packing 2 workers, each with 6GB of memory, onto each VM. It looks like this machine type would not be able to satisfy my requested amount of memory (16GB). In that case, what would happen? Would it fall back on the "guarantee" of 4GB? Or use as much as is available?

I am eager to get this merged and re-deploy so I can move ahead with my calculations.

**Comment by rabernat** [comment]
I'm stuck. The [update.md](https://github.com/pangeo-data/pangeo/blob/master/gce/update.md) document says:

> You will also need to update the tag of the pangeo/notebook docker image to the TAG that you created above

I can't find this `tag` entry in [jupyter-config.yaml](https://github.com/pangeo-data/pangeo/blob/master/gce/jupyter-config.yaml)

**Comment by jhamman** [comment]
@rabernat - I'm around. Off hand, I don't know where we specify the docker image for the notebook.

**Comment by rabernat** [comment]
@tjcrone is walking me through it right now

**Comment by mrocklin** [comment]
I'm around for a bit but not actively tracking issues/e-mail.  Feel free to
ping me on hangouts if you run into something.

**Comment by rabernat** [comment]
For those following this thread, I attempted to updated the cluster with the new chart and docker image. This is not working, as described in pangeo-data/helm-chart#19. The cluster is currently broken as a result.

**Comment by rabernat** [comment]
Even after attempting a rollback, this error is still present: `Failed to apply default image tag "None:None": couldn't parse image reference "None:None": invalid reference format: repository name must be lowercase: InvalidImageName`

So pangeo.pydata.org is now broken and I can't fix it.

**Comment by rabernat** [comment]
To get to the current state, I ran `helm rollback jupyter 2`

Here is the helm history
```
$ helm history jupyter
REVISION	UPDATED                 	STATUS    	CHART                	DESCRIPTION                                                 
1       	Thu Mar 29 08:31:07 2018	SUPERSEDED	pangeo-v0.1.0-673e876	Install complete                                            
2       	Thu Mar 29 09:05:32 2018	SUPERSEDED	pangeo-v0.1.0-673e876	Upgrade complete                                            
3       	Fri Apr 27 16:09:17 2018	FAILED    	pangeo-0.1.0-ef74d6d 	Upgrade "jupyter" failed: Deployment.apps "hub" is invali...
4       	Mon Apr 30 09:56:46 2018	SUPERSEDED	pangeo-v0.1.0-36c7a95	Upgrade complete                                            
5       	Mon Apr 30 10:31:07 2018	SUPERSEDED	pangeo-v0.1.0-36c7a95	Upgrade complete                                            
6       	Mon Apr 30 10:37:23 2018	SUPERSEDED	pangeo-v0.1.0-36c7a95	Upgrade complete                                            
7       	Mon Apr 30 10:56:35 2018	DEPLOYED  	pangeo-v0.1.0-673e876	Rollback to 2  
```

I sincerely do not understand how it is possible that I rolled back to a previously working deployment but now it still doesn't work.

**Comment by rabernat** [comment]
I somehow managed to coax the cluster back up. This process feels very shaky. I think that the problem is that, if we don't re-deploy for a month, kubernetes, helm, and gcloud all advance considerably and our config becomes out of date.

It appears that my new image requirements worked and the notebook pods have more memory. Yay! 👏 

Now I have hit a new problem: a bokeh incompatibility with the dask dashboard. Here is the error I got when I launched a KubeCluster
```
distributed.utils - ERROR - unexpected attribute 'index_position' to DataTable, possible attributes are columns, css_classes, disabled, editable, fit_columns, height, js_event_callbacks, js_property_callbacks, name, reorderable, row_headers, scroll_to_selection, selectable, sizing_mode, sortable, source, subscribed_events, tags, view or width
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/site-packages/distributed/utils.py", line 622, in log_errors
    yield
  File "/opt/conda/lib/python3.6/site-packages/distributed/bokeh/scheduler.py", line 1082, in workers_doc
    table = WorkerTable(scheduler)
  File "/opt/conda/lib/python3.6/site-packages/distributed/bokeh/scheduler.py", line 963, in __init__
    reorderable=True, sortable=True, width=width, **dt_kwargs
  File "/opt/conda/lib/python3.6/site-packages/bokeh/models/widgets/tables.py", line 493, in __init__
    super(TableWidget, self).__init__(**kw)
  File "/opt/conda/lib/python3.6/site-packages/bokeh/model.py", line 219, in __init__
    super(Model, self).__init__(**kwargs)
  File "/opt/conda/lib/python3.6/site-packages/bokeh/core/has_props.py", line 236, in __init__
    setattr(self, name, value)
  File "/opt/conda/lib/python3.6/site-packages/bokeh/core/has_props.py", line 271, in __setattr__
    (name, self.__class__.__name__, text, nice_join(matches)))
AttributeError: unexpected attribute 'index_position' to DataTable, possible attributes are columns, css_classes, disabled, editable, fit_columns, height, js_event_callbacks, js_property_callbacks, name, reorderable, row_headers, scroll_to_selection, selectable, sizing_mode, sortable, source, subscribed_events, tags, view or width
tornado.application - ERROR - Uncaught exception GET /workers (::1)
HTTPServerRequest(protocol='http', host='pangeo.pydata.org', method='GET', uri='/workers', version='HTTP/1.1', remote_ip='::1')
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/site-packages/tornado/web.py", line 1543, in _execute
    result = yield result
  File "/opt/conda/lib/python3.6/site-packages/tornado/gen.py", line 1099, in run
    value = future.result()
  File "/opt/conda/lib/python3.6/site-packages/tornado/gen.py", line 1107, in run
    yielded = self.gen.throw(*exc_info)
  File "/opt/conda/lib/python3.6/site-packages/bokeh/server/views/doc_handler.py", line 21, in get
    session = yield self.get_session()
  File "/opt/conda/lib/python3.6/site-packages/tornado/gen.py", line 1099, in run
    value = future.result()
  File "/opt/conda/lib/python3.6/site-packages/tornado/gen.py", line 1107, in run
    yielded = self.gen.throw(*exc_info)
  File "/opt/conda/lib/python3.6/site-packages/bokeh/server/views/session_handler.py", line 43, in get_session
    session = yield self.application_context.create_session_if_needed(session_id, self.request)
  File "/opt/conda/lib/python3.6/site-packages/tornado/gen.py", line 1099, in run
    value = future.result()
  File "/opt/conda/lib/python3.6/site-packages/tornado/gen.py", line 1113, in run
    yielded = self.gen.send(value)
  File "/opt/conda/lib/python3.6/site-packages/bokeh/server/application_context.py", line 182, in create_session_if_needed
    self._application.initialize_document(doc)
  File "/opt/conda/lib/python3.6/site-packages/bokeh/util/api.py", line 190, in wrapper
    return obj(*args, **kw)
  File "/opt/conda/lib/python3.6/site-packages/bokeh/application/application.py", line 179, in initialize_document
    h.modify_document(doc)
  File "/opt/conda/lib/python3.6/site-packages/bokeh/util/api.py", line 190, in wrapper
    return obj(*args, **kw)
  File "/opt/conda/lib/python3.6/site-packages/bokeh/application/handlers/function.py", line 131, in modify_document
    self._func(doc)
  File "/opt/conda/lib/python3.6/site-packages/distributed/bokeh/scheduler.py", line 1082, in workers_doc
    table = WorkerTable(scheduler)
  File "/opt/conda/lib/python3.6/site-packages/distributed/bokeh/scheduler.py", line 963, in __init__
    reorderable=True, sortable=True, width=width, **dt_kwargs
  File "/opt/conda/lib/python3.6/site-packages/bokeh/models/widgets/tables.py", line 493, in __init__
    super(TableWidget, self).__init__(**kw)
  File "/opt/conda/lib/python3.6/site-packages/bokeh/model.py", line 219, in __init__
    super(Model, self).__init__(**kwargs)
  File "/opt/conda/lib/python3.6/site-packages/bokeh/core/has_props.py", line 236, in __init__
    setattr(self, name, value)
  File "/opt/conda/lib/python3.6/site-packages/bokeh/core/has_props.py", line 271, in __setattr__
    (name, self.__class__.__name__, text, nice_join(matches)))
AttributeError: unexpected attribute 'index_position' to DataTable, possible attributes are columns, css_classes, disabled, editable, fit_columns, height, js_event_callbacks, js_property_callbacks, name, reorderable, row_headers, scroll_to_selection, selectable, sizing_mode, sortable, source, subscribed_events, tags, view or width
distributed.utils - ERROR - unexpected attribute 'index_position' to DataTable, possible attributes are columns, css_classes, disabled, editable, fit_columns, height, js_event_callbacks, js_property_callbacks, name, reorderable, row_headers, scroll_to_selection, selectable, sizing_mode, sortable, source, subscribed_events, tags, view or width
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/site-packages/distributed/utils.py", line 622, in log_errors
    yield
  File "/opt/conda/lib/python3.6/site-packages/distributed/bokeh/scheduler.py", line 1082, in workers_doc
    table = WorkerTable(scheduler)
  File "/opt/conda/lib/python3.6/site-packages/distributed/bokeh/scheduler.py", line 963, in __init__
    reorderable=True, sortable=True, width=width, **dt_kwargs
  File "/opt/conda/lib/python3.6/site-packages/bokeh/models/widgets/tables.py", line 493, in __init__
    super(TableWidget, self).__init__(**kw)
  File "/opt/conda/lib/python3.6/site-packages/bokeh/model.py", line 219, in __init__
    super(Model, self).__init__(**kwargs)
  File "/opt/conda/lib/python3.6/site-packages/bokeh/core/has_props.py", line 236, in __init__
    setattr(self, name, value)
  File "/opt/conda/lib/python3.6/site-packages/bokeh/core/has_props.py", line 271, in __setattr__
    (name, self.__class__.__name__, text, nice_join(matches)))
AttributeError: unexpected attribute 'index_position' to DataTable, possible attributes are columns, css_classes, disabled, editable, fit_columns, height, js_event_callbacks, js_property_callbacks, name, reorderable, row_headers, scroll_to_selection, selectable, sizing_mode, sortable, source, subscribed_events, tags, view or width
tornado.application - ERROR - Uncaught exception GET /workers (::1)
HTTPServerRequest(protocol='http', host='pangeo.pydata.org', method='GET', uri='/workers', version='HTTP/1.1', remote_ip='::1')
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/site-packages/tornado/web.py", line 1543, in _execute
    result = yield result
  File "/opt/conda/lib/python3.6/site-packages/tornado/gen.py", line 1099, in run
    value = future.result()
  File "/opt/conda/lib/python3.6/site-packages/tornado/gen.py", line 1107, in run
    yielded = self.gen.throw(*exc_info)
  File "/opt/conda/lib/python3.6/site-packages/bokeh/server/views/doc_handler.py", line 21, in get
    session = yield self.get_session()
  File "/opt/conda/lib/python3.6/site-packages/tornado/gen.py", line 1099, in run
    value = future.result()
  File "/opt/conda/lib/python3.6/site-packages/tornado/gen.py", line 1107, in run
    yielded = self.gen.throw(*exc_info)
  File "/opt/conda/lib/python3.6/site-packages/bokeh/server/views/session_handler.py", line 43, in get_session
    session = yield self.application_context.create_session_if_needed(session_id, self.request)
  File "/opt/conda/lib/python3.6/site-packages/tornado/gen.py", line 1099, in run
    value = future.result()
  File "/opt/conda/lib/python3.6/site-packages/tornado/gen.py", line 1113, in run
    yielded = self.gen.send(value)
  File "/opt/conda/lib/python3.6/site-packages/bokeh/server/application_context.py", line 182, in create_session_if_needed
    self._application.initialize_document(doc)
  File "/opt/conda/lib/python3.6/site-packages/bokeh/util/api.py", line 190, in wrapper
    return obj(*args, **kw)
  File "/opt/conda/lib/python3.6/site-packages/bokeh/application/application.py", line 179, in initialize_document
    h.modify_document(doc)
  File "/opt/conda/lib/python3.6/site-packages/bokeh/util/api.py", line 190, in wrapper
    return obj(*args, **kw)
  File "/opt/conda/lib/python3.6/site-packages/bokeh/application/handlers/function.py", line 131, in modify_document
    self._func(doc)
  File "/opt/conda/lib/python3.6/site-packages/distributed/bokeh/scheduler.py", line 1082, in workers_doc
    table = WorkerTable(scheduler)
  File "/opt/conda/lib/python3.6/site-packages/distributed/bokeh/scheduler.py", line 963, in __init__
    reorderable=True, sortable=True, width=width, **dt_kwargs
  File "/opt/conda/lib/python3.6/site-packages/bokeh/models/widgets/tables.py", line 493, in __init__
    super(TableWidget, self).__init__(**kw)
  File "/opt/conda/lib/python3.6/site-packages/bokeh/model.py", line 219, in __init__
    super(Model, self).__init__(**kwargs)
  File "/opt/conda/lib/python3.6/site-packages/bokeh/core/has_props.py", line 236, in __init__
    setattr(self, name, value)
  File "/opt/conda/lib/python3.6/site-packages/bokeh/core/has_props.py", line 271, in __setattr__
    (name, self.__class__.__name__, text, nice_join(matches)))
AttributeError: unexpected attribute 'index_position' to DataTable, possible attributes are columns, css_classes, disabled, editable, fit_columns, height, js_event_callbacks, js_property_callbacks, name, reorderable, row_headers, scroll_to_selection, selectable, sizing_mode, sortable, source, subscribed_events, tags, view or width
```

File: gce/notebook/Dockerfile [hunk 2]
```diff
@@ -8,7 +8,7 @@ RUN apt-get update \
 USER $NB_USER
 
 RUN conda install --yes -c ioam -c bokeh/channel/dev \
-    bokeh=0.12.15dev1 \
+    bokeh=0.12.15dev3 \
```
Comment (rabernat) [inline gce/notebook/Dockerfile]: I suppose this change is what introduced the bug. @mrocklin, is there a known dask dashboard incompatibility with this version of bokeh?

**Comment by rabernat** [comment]
Digging a bit deeper, it looks like I just found https://github.com/bokeh/bokeh/issues/7719

We will need to either roll back bokeh to an earlier version or else install from pip. I will defer to whatever @mrocklin recommends here.

After my previous experience, I am terrified of re-deploying the cluster again.

File: gce/notebook/Dockerfile (hunk 2 above)
Comment (mrocklin) [reply gce/notebook/Dockerfile]: Yes.  I recommend updating distributed to latest release to resolve this

**Comment by tjcrone** [comment]
In my efforts to get Pangeo running on Azure, unfortunately deploying dozens of times over several weeks, I ran into this sort of problem many times. I began to lock in version numbers for many packages into the Docker file, including Dask, distributed, bokeh, holoviews, jupyter, jupyterlab, and especially all the jupyterlab extensions once I had things working together. A lot of these projects are moving fast, but I figured that I would not need every new feature the moment one arrived. Specifically regarding bokeh/dask, I have had the best luck sticking with bokeh=0.12.14, dask=0.17.2 and distributed=1.21.4. Not sure if the newer dev versions are resolving something that you need resolved. Maybe I'm not running in to those issues. Or maybe I am and I don't know it.

File: gce/notebook/Dockerfile [hunk 3]
```diff
@@ -8,12 +8,12 @@ RUN apt-get update \
...
     cytoolz \
     datashader \
     dask=0.17.2 \
     dask-ml \
-    distributed=1.21.4 \
+    distributed=1.21.6 \
```
Comment (rabernat) [inline gce/notebook/Dockerfile]: @mrocklin - can you confirm that this version of distributed should work with `bokeh=0.12.15dev3`

**Comment by mrocklin** [comment]
Yes, should work

**Comment by rabernat** [comment]
Attempt to update cluster failed. See https://github.com/pangeo-data/helm-chart/issues/21

File: gce/jupyter-config.yaml [hunk 4]
```diff
@@ -57,3 +57,7 @@ jupyterhub:
     users: false
     timeout: 1200
     every: 600
+
+  proxy:
+    service:
+      loadBalancerIP: 35.224.8.169
```
Comment (rabernat) [inline gce/jupyter-config.yaml]: I put our new static IP in our site-specific config here. I based this on https://github.com/kubernetes/helm/issues/3560

Does this seem like the right thing to do?

FYI:
```
$ gcloud compute addresses list
NAME               REGION       ADDRESS       STATUS
pangeo-pydata-org  us-central1  35.224.8.169  IN_USE
```

Comment (rabernat) [reply gce/jupyter-config.yaml]: Note that we have to wait for Anaconda IT to update the pangeo.pydata.org DNS record to point to this IP. (Hopefully for the last time!)

Comment (rabernat) [reply gce/jupyter-config.yaml]: @yuvipanda or @choldgraf: is this the right way to bind our service to the static IP?

Comment (choldgraf) [reply gce/jupyter-config.yaml]: I think that's fine - we have mybinder.org use A records to point to our binderhub deployment on google cloud. Though I'll let @yuvipanda confirm this!

**Comment by mrocklin** [comment]
What is the status of this?

**Comment by rabernat** [comment]
The status is that we have deployed this PR many times already but have not yet merged it. The reason we have not merged it is because there is no clear way to decide when something like this should be merged. It now appears to be in conflict with your latest updates.

When you last deployed the cluster, did you use the increased image memory settings from this PR?

**Comment by rabernat** [comment]
Also, I was hoping to resolve #236 before merging.

**Comment by mrocklin** [comment]
> When you last deployed the cluster, did you use the increased image memory settings from this PR?

Nope, I was just modifying from master.

**Comment by rabernat** [comment]
Ok so then we need to resolve and merge this. I will do that now.

**Comment by rabernat** [comment]
@mrocklin: can you tell me the latest command you used to upgrade the cluster? (We had some trouble with `helm upgrade` recently.) I will attempt to upgrade to latest master now that this is merged.

**Comment by rabernat** [comment]
Latest helm chart `0.1.1-08b10bd` has been built: https://pangeo-data.github.io/helm-chart/

**Comment by mrocklin** [comment]
Woot.  Thanks for sorting through that @rabernat

**Comment by rabernat** [comment]
Yet there still seems to be a problem:
```
$ helm inspect pangeo/pangeo
Error: failed to download "pangeo/pangeo"
```

**Comment by mrocklin** [comment]
helm repo update ?

**Comment by mrocklin** [comment]
Ah no, I'm getting the same thing

**Comment by rabernat** [comment]
I'll see what happens if I try to upgrade.

**Comment by rabernat** [comment]
Upgrade appears to have succeeded. However, there is a weird port forwarding error message. I don't know if this is a cause for concern. 

```
$ helm upgrade jupyter pangeo/pangeo --version 0.1.1-08b10bd -f jupyter-config.yaml -f secret-config.yaml
Release "jupyter" has been upgraded. Happy Helming!
E0507 13:25:52.362751    1733 portforward.go:303] error copying from remote stream to local connection: readfrom tcp4 127.0.0.1:52847->127.0.0.1:52850: write tcp4 127.0.0.1:52847->127.0.0.1:52850: write: broken pipe
LAST DEPLOYED: Mon May  7 13:23:03 2018
NAMESPACE: pangeo
STATUS: DEPLOYED

RESOURCES:
==> v1/PersistentVolumeClaim
NAME        STATUS  VOLUME                                    CAPACITY  ACCESS MODES  STORAGECLASS  AGE
hub-db-dir  Bound   pvc-78f2badc-4d78-11e8-b35e-42010a800fc2  1Gi       RWO           standard      5d

==> v1/ServiceAccount
NAME            SECRETS  AGE
hub             1        5d
daskkubernetes  1        5d

==> v1beta1/Role
NAME            AGE
hub             5d
daskkubernetes  5d

==> v1beta1/Deployment
NAME   DESIRED  CURRENT  UP-TO-DATE  AVAILABLE  AGE
hub    1        1        1           0          5d
proxy  1        1        1           1          5d

==> v1beta1/PodDisruptionBudget
NAME   MIN AVAILABLE  MAX UNAVAILABLE  ALLOWED DISRUPTIONS  AGE
hub    1              N/A              0                    5d
proxy  1              N/A              0                    5d

==> v1/Secret
NAME        TYPE    DATA  AGE
hub-secret  Opaque  1     5d

==> v1/ConfigMap
NAME        DATA  AGE
hub-config  36    5d

==> v1beta1/RoleBinding
NAME            AGE
hub             5d
daskkubernetes  5d

==> v1/Service
NAME          TYPE          CLUSTER-IP     EXTERNAL-IP   PORT(S)       AGE
hub           ClusterIP     10.23.250.162  <none>        8081/TCP      5d
proxy-api     ClusterIP     10.23.252.8    <none>        8001/TCP      5d
proxy-public  LoadBalancer  10.23.255.86   35.224.8.169  80:30686/TCP  5d

==> v1/Pod(related)
NAME                    READY  STATUS             RESTARTS  AGE
hub-6685874d8b-d8b2j    1/1    Terminating        0         18h
hub-6f4b7c7b44-6tdcz    0/1    ContainerCreating  0         0s
proxy-64f95bb784-hjdxc  1/1    Running            0         18h


NOTES:
Pangeo
------

Thank you for installing Pangeo!

Your release is named jupyter and installed into the namespace pangeo.


Jupyter Hub
-----------

You can find if the hub and proxy is ready by doing:

 kubectl --namespace=pangeo get pod

and watching for both those pods to be in status 'Ready'.

You can find the public IP of the JupyterHub by doing:

 kubectl --namespace=pangeo get svc proxy-public

It might take a few minutes for it to appear!


Support
-------

Note that this is still an alpha release! If you have questions, feel free to file issues at https://github.com/pangeo-data/pangeo/issues.
```

**Comment by jacobtomlinson** [comment]
I see that port forwarding error intermittently when using helm generally. I haven't noticed any issues caused by it and tend to just run the command again to be sure.

**Pull request merged**